airflow/models/dagrun.py

```diff
@@ -286,7 +287,17 @@ def find(
     @staticmethod
     def generate_run_id(run_type: DagRunType, execution_date: datetime) -> str:
         """Generate Run ID based on Run Type and Execution Date"""
-        return f"{run_type}__{execution_date.isoformat()}"
+        from airflow.configuration import conf
+        local_run_id = conf.getboolean("core", "localize_dag_run_id", fallback=False)
+        if local_run_id:
+            local_time = pendulum.instance(execution_date).astimezone(tz=settings.TIMEZONE)
+            is_dst = local_time.is_dst()
+            if is_dst:
+                return f"{run_type}__{local_time.isoformat()}__DST"
```

Comment:
I don't see the need for this suffix -- when it's in DST the timezone offset will already be different (`+00:00` vs `+01:00__DST` in your example for UK)

Reply:
Actually, I don't know the living habits who are under DST, because we don't use the DST in China.
I just think it will be much clearer if we add `__DST`.
I will remove it tomorrow